packages/vite-plugin-checker/src/checkers/typescript/main.ts

```diff
@@ -120,7 +120,14 @@ export class TscChecker extends Checker<'typescript'> {
     super({
       name: 'typescript',
       absFilePath: __filename,
-      build: { buildBin: ['tsc', ['--noEmit']] },
+      build: {
+        buildBin: (config) => {
+          if (typeof config.typescript === 'object' && config.typescript.tsconfigPath) {
+            return ['tsc', ['--noEmit', '-p', config.typescript.tsconfigPath]]
```

Comment:
should we also confider `config.typescript.root` and join it with `config.typescript.tsconfigPath` for the finalized tsconfig path

Reply:
I can do that, the only thing is that it might be a little confusing as the `tsc` process is always going to be spawned with a working directory of `process.cwd()` so it won't actually run in the `root` folder but will look for the `tsconfig` there. I'm not sure whether that's a problem or not.

Reply:
I added breakpoint and run `tsc -p SOME_ABS_TSCONFIG_PATH` and looked into the config. Seems like if the config file is specified as an absolute path explicitly, the cwd will be ignored. I looked into typescript source code I guess relative path config will turn into the absolute path in the end. 

https://github.com/microsoft/TypeScript/blob/971133d5d0a56cf362571d21ac971888f8a66820/src/executeCommandLine/executeCommandLine.ts#L503

And I tried in some projects in different roots with an absolute path tsconfig. Things work fine. So I think we can do that.

Reply:
I changed it so it also takes `root` into account. I think the nicest solution would actually be to mirror the behavior in `configureServer` with `ts.findConfigFile` but that would require more drastic changes to how checkers are created (they would need to access the project root directory, as configured in Vite). I think the current solution should work fine for most cases though, especially when using absolute paths for the config files.

Reply:
Great!